src/shogun/base/SGObject.h

```diff
@@ -966,6 +938,27 @@ class CSGObject
 	 */
 	void observe(const Some<ObservedValue> value);
 
+	/**
+	 * Observe a parameter value given some information
+	 * @tparam T value of the parameter
+	 * @param step step
+	 * @param name name of the observed value
+	 * @param description description
+	 * @param value observed value
+	 */
+	template <class T>
+	void
+	observe(int64_t step, std::string name, std::string description, T value);
+
+	/**
+	 * Observe a registered tag.
+	 * @tparam T type of the tag
+	 * @param step step
+	 * @param name tag's name
+	 */
+	template <class T>
+	void observe(int64_t step, std::string name);
```

Comment:
not sure why we would need this one? To avoid passing the value? Is that an advantage?